packages/data-lake-api/src/SqlQuery.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+export class SqlQuery {
+  static array = arr => `(${arr.map(() => '?').join(',')})`;
+
+  static values = rows =>
+    `VALUES (${rows.map(values => values.map(() => `?`).join(',')).join('), (')})`;
+
+  static innerJoin = (baseTable, columnName, values) => `
+    INNER JOIN (
+      ${SqlQuery.values(values.map(c => [c]))}
+    ) ${columnName}s(code) ON ${columnName}s.code = ${baseTable}.${columnName}
+  `;
+
+  constructor(baseQuery, baseParameters = []) {
+    this.query = baseQuery;
+    this.parameters = baseParameters;
+  }
+
+  async executeOnDatabase(database) {
+    return database.executeSql(this.query, this.parameters);
+  }
+
+  loggableQuery() {
```

Comment:
Knex has query logging built in :) 
Otherwise something like this works wonders https://github.com/beyondessential/tupaia/blob/dev/packages/database/src/TupaiaDatabase.js#L516

Reply:
This is all copy pasted from the `SqlQuery.js` in `@tupaia/data-api`, however I think in my experience with `query.toString()` it didn't handle logging query parameters correctly, it just left them as `?`... I'll have to double check that now as my memory is a bit foggy on it

Reply:
No worries, minor comment